src/InclusiveScans.jl

```diff
@@ -1,12 +1,166 @@
 module InclusiveScans
 
+using CUDA
+using CUDA: i32
 
-"""
-    hi = hello_world()
-A simple function to return "Hello, World!"
-"""
-function hello_world()
-    return "Hello, World!"
+const BLOCK_SIZE::Int32 = 1024
+
+function _scanBlockKernel!(
+    ::Type{T},
+    g_odata,
+    g_idata,
+    blockSums,
+    n::Int32,
+) where {T}
+    temp = CuDynamicSharedArray(T, blockDim().x * 2)
+
+    tx = threadIdx().x - 1i32
+    bx = blockIdx().x - 1i32
+
+    start = bx * (blockDim().x * 2)
+
+    i1 = start + tx * 2
```

Comment:
Each place, where address calculation is done is affected by problem, that a constant number is 64 bit and needs to be extend by the `i32` suffix.